traffic.lua

```diff
@@ -282,4 +289,172 @@ function courseplay:deleteCollisionVehicle(vehicle)
+			vehicle.aiTrafficCollisionTrigger = getChild(vehicle.rootNode, "aiCollisionTrigger");
+		else
+			for i=0,getNumOfChildren(vehicle.rootNode)-1 do
+				local child = getChildAt(vehicle.rootNode, i);
+				if getChild(child, "aiCollisionTrigger") ~= 0 then
+					vehicle.aiTrafficCollisionTrigger = getChild(child, "aiCollisionTrigger");
+					if vehicle.aiTrafficCollisionTrigger then
+						-- CpManager.trafficCollisionIgnoreList[vehicle.aiTrafficCollisionTrigger] = true; --add AI traffic collision trigger to global ignore list
+						break;
+					end
+				end;
+			end;
+		end;
+	end;
+
+	if vehicle.aiTrafficCollisionTrigger == nil then
+		print(string.format('## Courseplay: aiTrafficCollisionTrigger missing. Traffic collision prevention will not work! vehicle %s', nameNum(vehicle)));
+	end;
+
+	if vehicle.aiTrafficCollisionTrigger then
+		-- CpManager.trafficCollisionIgnoreList[vehicle.aiTrafficCollisionTrigger] = true; --add AI traffic collision trigger to global ignore list
+		ret = true;
+	end
+
+	return ret;
+end
+
+function courseplay:createLegacyCollisionTriggers(vehicle)
+	if vehicle == nil then 
+		return false;
+	end;
+
+	local ret = false
+	
+	if vehicle.cp.trafficCollisionTriggers[1]	== nil then
+		--TODO Tommi: remove this when we are completely on AIDriver
+		vehicle.cp.trafficCollisionTriggerToTriggerIndex = {};
+		if vehicle.aiTrafficCollisionTrigger ~= nil then
+			for i=1,vehicle.cp.numTrafficCollisionTriggers do
+				local newTrigger = clone(vehicle.aiTrafficCollisionTrigger, true);
+				vehicle.cp.trafficCollisionTriggers[i] = newTrigger
+				if i > 1 then
+					unlink(newTrigger)
+					link(vehicle.cp.trafficCollisionTriggers[i-1], newTrigger);
+					setTranslation(newTrigger, 0,0,5);
+				end;
+				addTrigger(newTrigger, 'cpOnTrafficCollisionTrigger', vehicle);
+				vehicle.cp.trafficCollisionTriggerToTriggerIndex[newTrigger] = i;
+				-- CpManager.trafficCollisionIgnoreList[newTrigger] = true; --add all traffic collision triggers to global ignore list
+				vehicle.cp.collidingObjects[i] = {};
+				ret = true
+			end;
+		end;
+	end;
+	return ret;
+end
+
+function courseplay:removeLegacyCollisionTriggers(vehicle)
+	if vehicle == nil then
+		return false;
+	end;
+
+	local ret = false
+
+	--TODO Tommi: remove this when we are completely on AIDriver
+	if vehicle.cp.trafficCollisionTriggers[1] ~= nil then
+		for i=vehicle.cp.numTrafficCollisionTriggers,1,-1 do 
+			local node = vehicle.cp.trafficCollisionTriggers[i]
+			if node then
+				removeTrigger(node)
+				if entityExists(node) then
+					unlink(node)
+					vehicle:removeWashableNode(node)
+					vehicle:removeWearableNode(node)
+					delete(node)
+				end
+			end
+			-- CpManager.trafficCollisionIgnoreList[node] = nil
+			vehicle.cp.collidingObjects[i] = {};
+			vehicle.cp.trafficCollisionTriggers[i] = nil
+			ret = true
+		end
+	end;
+	return ret;
+end
+
+function courseplay:setTrafficCollision_onfield(vehicle, lx, lz, disableLongCheck)
```

Comment:
Can we please use lowerCamelCase

Reply:
Do you mean "setTrafficCollision_onField" ?

Reply:
setTrafficCollisionOnField

Reply:
Done.